apps/insights/src/components/MobileMenu/mobile-menu.v2.tsx

```diff
@@ -0,0 +1,61 @@
+'use client';
+import { Lifebuoy } from "@phosphor-icons/react/dist/ssr/Lifebuoy";
+import { List } from "@phosphor-icons/react/dist/ssr/List";
+import { X } from "@phosphor-icons/react/dist/ssr/X";
+import { Button } from "@pythnetwork/component-library/Button";
+import clsx from "clsx";
+import { useTheme } from "next-themes";
+import { useState } from "react";
+
+import styles from "./mobile-menu.module.scss";
+import { SupportDrawer } from "../Root/support-drawer";
+import { ThemeSwitch } from "../Root/theme-switch";
+
+
+export const MobileMenu = ({
+    className,
+    ...props
+}: {
+    className?: string | string[];
+}) => {
+    const { theme } = useTheme();
+    const [isOpen, setIsOpen] = useState(false);
+
+    const toggleMenu = () => {
+        setIsOpen(!isOpen);
+    };
+
+    return (
+        <div className={clsx(styles.mobileMenuTrigger, className)} {...props}>
```

Comment:
Rather than manually coding out this overlay, we should be using the `ModalDialog` component (or one of the components that use it like `Drawer`) from the component library.  The `ModalDialog` will handle a ton of accessibility issues like tab locking, setting proper aria attributes, focus control, and will also apply animation in/out states automatically for you.  It should generally be used any time you are coding up something that is modal and overlays the content of the page.